src/sonic-config-engine/sonic_device_util.py

```diff
@@ -27,7 +30,45 @@ def get_machine_info():
             if len(tokens) < 2:
                 continue
             machine_vars[tokens[0]] = tokens[1].strip()
-    return machine_vars
+    return machine_vars 
+
+def get_npu_id_from_name(npu_name):
+    if npu_name.startswith(NPU_NAME_PREFIX):
+        return npu_name[len(NPU_NAME_PREFIX):]
+    else:
+        return None
+
+def get_num_npus():
+   platform = get_platform_info(get_machine_info())
+   asic_conf_file_path = os.path.join(SONIC_DEVICE_PATH, platform, 'asic.conf')
+   if not os.path.isfile(asic_conf_file_path):
+        return 1
+   with open(asic_conf_file_path) as asic_conf_file:
+	for line in asic_conf_file:
+	    tokens = line.split('=')
+            if len(tokens) < 2:
+               continue      
+	    if tokens[0].lower() == 'num_asic':
+		num_npus = tokens[1].strip()
+   		return num_npus
+
+def get_namespaces():
+    """
+    In a multi NPU platform, each NPU is in a Linux Namespace.
+    This method returns list of all the Namespace present on the device
+    """
+    ns_list = []
+    try:
+        proc = subprocess.Popen('ip netns list | cut -d"(" -f1', 
+                                stdout=subprocess.PIPE,
+                                shell=True,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        ns_list = [n for n in stdout.split()]
+    except OSError,e:
+        raise OSError("Unable to get namespace list")
+    return natsorted(ns_list)
```

Comment:
no need to sort here. you do not know what order the user wants

Reply:
This function is used mainly by CLI, in that case it is better if namespace list is sorted